jitbuilder/runtime/JBCodeCacheMemorySegment.hpp

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2023
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+
+#ifndef JITBUILDER_CODECACHEMEMORYSEGMENT_INCL
+#define JITBUILDER_CODECACHEMEMORYSEGMENT_INCL
+
+#ifndef JITBUILDER_CODECACHEMEMORYSEGMENT_COMPOSED
+#define JITBUILDER_CODECACHEMEMORYSEGMENT_COMPOSED
+namespace JitBuilder { class CodeCacheMemorySegment; }
+namespace JitBuilder { typedef CodeCacheMemorySegment CodeCacheMemorySegmentConnector; }
+#endif
+
+#include "runtime/OMRCodeCacheMemorySegment.hpp"
+
+namespace TR { class CodeCacheMemorySegment; }
+
+namespace JitBuilder
+{
+
+class OMR_EXTENSIBLE CodeCacheMemorySegment : public OMR::CodeCacheMemorySegmentConnector
+   {
+   public:
+   CodeCacheMemorySegment()                              : OMR::CodeCacheMemorySegmentConnector()             { }
+   CodeCacheMemorySegment(uint8_t *memory, size_t size)  : OMR::CodeCacheMemorySegmentConnector(memory, size) { }
+   CodeCacheMemorySegment(uint8_t *memory, uint8_t *top) : OMR::CodeCacheMemorySegmentConnector(memory, top)  { }
+
+   TR::CodeCacheMemorySegment *self();
+
+   void setSegmentBase(uint8_t *newBase);
+   void setSegmentAlloc(uint8_t *newAlloc);
+   void setSegmentTop(uint8_t *newTop);
+   };
+
+}
+
+#endif // defined(JITBUILDER_CODECACHEMEMORYSEGMENT_INCL)
```

Comment:
The comment here has to be `!defined(JITBUILDER_CODECACHEMEMORYSEGMENT_INCL)`.

Reply:
Updated the comment.